client/src/components/DataFiles/DataFilesModals/DataFilesLinkModal.js

```diff
@@ -0,0 +1,188 @@
+  method: PropTypes.string.isRequired,
+  status: statusPropType
+};
+
+DataFilesLinkAction.defaultProps = {
+  status: {
+    error: null,
+    url: null,
+    method: null
+  }
+};
+
+const DataFilesLinkStatus = ({ scheme, file, status }) => {
+  if (!status) {
+    return null;
+  }
+  if (status && status.error) {
+    // Error occurred during retrieval of link
+    return (
+      <Message type="error">
+        There was a problem retrieving the link for this file.
+      </Message>
+    );
+  }
+  return (
+    <FormGroup>
+      <Label>
+        Link
+        {status && status.method === 'get' ? (
+          <LoadingSpinner placement="inline" />
+        ) : null}
+      </Label>
+      <TextCopyField
+        placeholder="Click generate to make a link"
+        value={status.url}
+      />
+      <div styleName="controls">
+        {status && status.url ? (
+          <DataFilesLinkAction
+            scheme={scheme}
+            file={file}
+            text="Delete"
+            status={status}
+            method="delete"
+          />
+        ) : null}
+        <DataFilesLinkAction
+          scheme={scheme}
+          file={file}
+          text={status && status.url ? 'Replace Link' : 'Generate Link'}
+          status={status}
+          method="post"
+        />
+      </div>
+    </FormGroup>
+  );
+};
+
+DataFilesLinkStatus.propTypes = {
+  scheme: PropTypes.string.isRequired,
+  file: filePropType.isRequired,
+  status: statusPropType
+};
+
+DataFilesLinkStatus.defaultProps = {
+  status: null
+};
+
+const DataFilesLinkModal = () => {
+  const isOpen = useSelector(state => state.files.modals.link);
+  const status = useSelector(state => state.files.operationStatus.link);
+  const { scheme } = useSelector(state => state.files.params.FilesListing);
+  const selectedFile = useSelector(state => {
+    if (!state.files.modalProps.link) {
+      return {};
+    }
+    return state.files.modalProps.link.selectedFile || {};
+  });
+
+  const dispatch = useDispatch();
+  const toggle = () => {
+    dispatch({
+      type: 'DATA_FILES_TOGGLE_MODAL',
+      payload: { operation: 'link', props: {} }
+    });
+  };
+
+  const onClosed = () => {
+    dispatch({
+      type: 'DATA_FILES_SET_OPERATION_STATUS',
+      payload: { status: null, operation: 'link' }
+    });
+  };
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClosed={onClosed}
+      toggle={toggle}
+      className="dataFilesModal"
+    >
+      <Form>
+        <ModalHeader toggle={toggle}>Link for {selectedFile.name}</ModalHeader>
+        <ModalBody>
+          <DataFilesLinkStatus
+            scheme={scheme}
+            file={selectedFile}
+            status={status}
+          />
+        </ModalBody>
+        <ModalFooter>
+          <Button
+            color="secondary"
+            className="data-files-btn-cancel"
+            onClick={toggle}
+          >
+            Close
+          </Button>
+        </ModalFooter>
```

Comment:
think we're nixing close buttons
```suggestion
        <ModalFooter>
          <Button
            color="secondary"
            className="data-files-btn-cancel"
            onClick={toggle}
          >
            Close
          </Button>
        </ModalFooter>
```